### PR TITLE
CLOUD-55526 armTemplate clean-up

### PR DESCRIPTION
**Pull request by bioneyez**
@doktoric pls review

### CODE REVIEW

**Comment by doktoric** [comment]
MUST BE TESTED BEFORE MERGE

**Comment by mhmxs** [comment]
I would test it, see you soon :)

File: cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmTemplateBuilder.java [hunk 1]
```diff
@@ -77,6 +77,28 @@ public String build(String stackName, ArmCredentialView armCredentialView, ArmSt
             model.put("resourceGroupName", armUtils.getCustomResourceGroupName(network));
             model.put("existingVNETName", armUtils.getCustomNetworkId(network));
             model.put("existingSubnetName", armUtils.getCustomSubnetId(network));
+            model.put("userImageName", String.format("https://%s.blob.core.windows.net/%s/%s", rootDiskStorage, ArmStorage.IMAGES, imageName));
+            model.put("osDiskVhdName", String.format("https://%s.blob.core.windows.net/%s/%sosDisk",
+                    rootDiskStorage, armStorage.getDiskContainerName(cloudContext), stackName));
+            model.put("sshKeyPath", String.format("/home/%s/.ssh/authorized_keys", armCredentialView.getLoginUserName()));
+            model.put("ipConfigurationsAddress", String.format("'/frontendIPConfigurations/%sipcn", stackName));
```
Comment (mhmxs) [inline cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmTemplateBuilder.java]: I guess ' character should occur template validation error near `'${ipConfigurationsAddress}'`

File: cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmTemplateBuilder.java [hunk 2]
```diff
@@ -77,6 +77,28 @@ public String build(String stackName, ArmCredentialView armCredentialView, ArmSt
...
+            model.put("ilbBackendAddress", String.format("/backendAddressPools/%sbapn", stackName));
+            model.put("loadBalancerAddress", String.format("Microsoft.Network/loadBalancers/%slb", stackName));
+            model.put("virtualNetworkAddress", String.format("Microsoft.Network/virtualNetworks/%s", stackName));
+            model.put("publicIpAddressId", String.format("Microsoft.Network/publicIPAddresses/%s", stackName));
+            model.put("networkInterfaceAddress", String.format("Microsoft.Network/networkInterfaces/%s", stackName));
```
Comment (mhmxs) [inline cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmTemplateBuilder.java]: There is one more place where interface address is replaceable: https://github.com/bioneyez/cloudbreak/blob/d3b5b53e9247325b655e2541f1591233987eccaf/cloud-arm/src/main/resources/templates/arm-v2.ftl#L359
Here we can use stackname: https://github.com/bioneyez/cloudbreak/blob/d3b5b53e9247325b655e2541f1591233987eccaf/cloud-arm/src/main/resources/templates/arm-v2.ftl#L239
And than we don't need this one: https://github.com/bioneyez/cloudbreak/blob/d3b5b53e9247325b655e2541f1591233987eccaf/cloud-arm/src/main/resources/templates/arm-v2.ftl#L52

File: cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmTemplateBuilder.java [hunk 3]
```diff
@@ -77,6 +77,28 @@ public String build(String stackName, ArmCredentialView armCredentialView, ArmSt
...
             model.put("existingVNETName", armUtils.getCustomNetworkId(network));
             model.put("existingSubnetName", armUtils.getCustomSubnetId(network));
+            model.put("userImageName", String.format("https://%s.blob.core.windows.net/%s/%s", rootDiskStorage, ArmStorage.IMAGES, imageName));
+            model.put("osDiskVhdName", String.format("https://%s.blob.core.windows.net/%s/%sosDisk",
+                    rootDiskStorage, armStorage.getDiskContainerName(cloudContext), stackName));
+            model.put("sshKeyPath", String.format("/home/%s/.ssh/authorized_keys", armCredentialView.getLoginUserName()));
+            model.put("ipConfigurationsAddress", String.format("'/frontendIPConfigurations/%sipcn", stackName));
+            model.put("ilbBackendAddress", String.format("/backendAddressPools/%sbapn", stackName));
+            model.put("loadBalancerAddress", String.format("Microsoft.Network/loadBalancers/%slb", stackName));
+            model.put("virtualNetworkAddress", String.format("Microsoft.Network/virtualNetworks/%s", stackName));
```
Comment (mhmxs) [inline cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmTemplateBuilder.java]: The name prefix replaceable with stackname here: https://github.com/bioneyez/cloudbreak/blob/d3b5b53e9247325b655e2541f1591233987eccaf/cloud-arm/src/main/resources/templates/arm-v2.ftl#L109
And we don't need this any more: https://github.com/bioneyez/cloudbreak/blob/d3b5b53e9247325b655e2541f1591233987eccaf/cloud-arm/src/main/resources/templates/arm-v2.ftl#L39

Comment (mhmxs) [inline cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmTemplateBuilder.java]: As i see this variable needs only if !existingVPC: https://github.com/bioneyez/cloudbreak/blob/d3b5b53e9247325b655e2541f1591233987eccaf/cloud-arm/src/main/resources/templates/arm-v2.ftl#L248

**Comment by mhmxs** [comment]
Never tested. The result is an invalid template.

**Pull request closed without merging**